website/static/js/fangorn.js

```diff
@@ -1061,12 +1654,35 @@ tbOptions = {
         $(document).on('click', '.fangorn-dismiss', function() {
             tb.redraw();
         });
+        tb.select('#tb-tbody').on('click', function(event){
+            if(event.target !== this) {
+                return;
+            }
+            tb.clearMultiselect();
+            _fangornResetToolbar.call(tb);
+        })
 
         $(window).on('beforeunload', function() {
             if (tb.dropzone && tb.dropzone.getUploadingFiles().length) {
               return 'You have pending uploads, if you leave this page they may not complete.';
             }
         });
+        if(tb.options.placement === 'project-files') {
+            _resizeHeight.call(tb);
+            $(window).resize(function(){
+                _resizeHeight.call(tb);
+            })
+        }
+        $(window).on('keydown', function(event){
+            if (event.keyCode === 27) {
+                _fangornResetToolbar.call(tb);
+            }
+        });
+        $(document).on('keypress', '#createFolderInput', function () {
+            if (tb.pressedKey === 13) {
```

Comment:
Treebeard is setting a state outside of the immeadiate event because move related functiosn need to know if at that point a key is being pressed, that's how they know whether to cop or move for instance. So that makes sure the key press state is always available and fangorn is just using that since it's available.